ringbuffer.py

```diff
@@ -93,15 +97,16 @@ class RingBuffer:
     will not work.
     """
 
-    def __init__(self, *, slot_bytes, slot_count):
+    def __init__(self, *, c_type: Type[ctypes._SimpleCData], slot_count: int):
```

Comment:
Hm... my suspicion is that we should be using a generic typevar with an upper bound (something like `T = TypeVar("T", bound=ctypes._SimpleCData)`.